src/serac/physics/solid_functional.hpp

```diff
@@ -0,0 +1,579 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file solid_functional.hpp
+ *
+ * @brief An object containing the solver for total Lagrangian finite deformation solid mechanics
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+#include "serac/physics/base_physics.hpp"
+#include "serac/numerics/odes.hpp"
+#include "serac/numerics/stdfunction_operator.hpp"
+#include "serac/numerics/functional/functional.hpp"
+#include "serac/physics/state/state_manager.hpp"
+#include "serac/physics/solid.hpp"
+#include "serac/physics/materials/functional_material_utils.hpp"
+
+namespace serac {
+
+/**
+ * @brief The nonlinear solid solver class
+ *
+ * The nonlinear total Lagrangian quasi-static and dynamic
+ * hyperelastic solver object. This uses Functional to compute the tangent
+ * stiffness matrices.
+ *
+ * @tparam order The order of the discretization of the displacement and velocity fields
+ * @tparam dim The spatial dimension of the mesh
+ */
+template <int order, int dim>
+class SolidFunctional : public BasePhysics {
```

Comment:
If Functional is the future, should we rename this to just `Solid` and make the old one `SolidDontUseThisBecauseTheNewWayIsSuperCool` though I am only seeing a `ThermalSolid` class.

Reply:
Yep - that's on the TODO list probably in the very near future once key customers switch over (cough cough @sethwatts 😆 )